src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -383,11 +391,11 @@ public void CreateCustomCharacter(byte location, params byte[] characterMap)
         /// <param name="characterMap">Provide an array of 8 bytes containing the pattern</param>
         public void CreateCustomCharacter(byte location, ReadOnlySpan<byte> characterMap)
         {
-            if (location > 7)
+            if (location > NumberOfCustomCharactersSupported - 1)
                 throw new ArgumentOutOfRangeException(nameof(location));
 
             if (characterMap.Length != 8)
-                throw new ArgumentException(nameof(characterMap));
+                throw new ArgumentException("The character map must be exactly 8 bytes long", nameof(characterMap));
```

Comment:
```suggestion
                throw new ArgumentException($"The character map must be exactly {characterMap.Length} bytes long", nameof(characterMap));
```

Reply:
Almost... Your suggestion would always print the actual value, not the expected. 
Added a constant.